### PR TITLE
Text summarization app | Lev Golofastov

### PR DESCRIPTION
**Pull request by lev-golofastov**
Simple Streamlit application for summarizing the given text.
Created as a homework for practice of programming at SPbU by Lev Golofastov.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import streamlit as st
+import torch
+from transformers import pipeline
+
+
+def analyze_text(text_to_analyze):
+	summarizer = pipeline(
```
Comment (SpirinEgor) [inline src/main.py:7]: What padding do you use? It seems that there are 8 spaces instead of the classic 4 🧐

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		"summarization",
+		"pszemraj/long-t5-tglobal-base-16384-book-summary",
+		device=0 if torch.cuda.is_available() else -1,
```
Comment (SpirinEgor) [inline src/main.py:10]: Use `"cuda"` or `"cpu"` for the device name.

```python
device = "cuda" if torch.cuda.is_available() else "cpu"
```

File: src/main.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	)
+
+	result = summarizer(text_to_analyze)
```
Comment (SpirinEgor) [inline src/main.py:13]: You can customize results by passing `min_length`, `max_length`, `beam_search`, etc. See [`#generate`](https://huggingface.co/docs/transformers/main_classes/text_generation) for all parameters.

You can add sliders in the interface, so a user may set up a generation algorithm.

File: src/main.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	st.write("Model source: https://huggingface.co/pszemraj/long-t5-tglobal-base-16384-book-summary")
+
+	sample = ""
+	with open('sample.txt') as f:
+		lines = f.readlines()
+		for line in lines:
+			sample += line
+
+	st.write("A sample text for testing, found on model\'s web page:")
+	st.write(sample)
```
Comment (SpirinEgor) [inline src/main.py:32]: You can specify this text as a placeholder for the text area. So the user can either generate a summary for it or replace it with a new one.

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
Good, but let's improve interface for better usage experience!

File: src/main.py (hunk 2 above)
Comment (lev-golofastov) [reply src/main.py]: Fix in cecca7f

File: src/main.py (hunk 4 above)
Comment (lev-golofastov) [reply src/main.py]: Fix in 3f6a369

File: src/main.py (hunk 1 above)
Comment (lev-golofastov) [reply src/main.py]: Fix in daf1c2f

File: src/main.py (hunk 3 above)
Comment (lev-golofastov) [reply src/main.py]: Add in 52e61b3

File: src/main.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def analyze_text(text_to_analyze, min_length, max_length):
+    summarizer = pipeline(
+        "summarization",
+        "pszemraj/long-t5-tglobal-base-16384-book-summary",
+        device="cuda" if torch.cuda.is_available() else "cpu",
+        min_length=min_length,
+        max_length=max_length
+    )
```
Comment (SpirinEgor) [inline src/main.py:13]: This will reload the pipeline on each generation. Try to avoid this in future work.

**Review by SpirinEgor — APPROVED** [review]
👍

File: src/main.py (hunk 5 above)
Comment (lev-golofastov) [reply src/main.py:13]: Thank you! I will try my best :)
